### PR TITLE
DEV-523: shared print policies; commitment tsv loading

### PR DESCRIPTION
**Pull request by aelkiss**
@mwarin  I think this covers the needs of DEV-523 as I understand them:

- validate policies being a specific list
- allow TSV loading of commitments, including policies

It does NOT:
- normalize other fields coming in to lowercase (either for TSV or JSON), in particular local shelving type
- handle the alleged "Do not reproduce" policy (I'm not sure how that would be formatted in practice)

Please feel free either to kick it back to me or take it over if there are additional needs.

### CODE REVIEW

File: lib/loader/file_loader.rb [hunk 1]
```diff
@@ -36,7 +36,8 @@ def skip_matching_header(filehandle, skip_match)
     # @return [Integer] Number of records loaded
     def load(filename, filehandle: Zinzout.zin(filename), skip_header_match: nil)
       logger.info "Loading #{filename}, batches of #{ppnum @batch_size}"
-      skip_matching_header(filehandle, skip_header_match).each.lazy
+
+      skip_matching_header(filehandle.enum_for(:each), skip_header_match).lazy
```
Comment (aelkiss) [inline lib/loader/file_loader.rb:40]: this change ensures we get an enumerator back regardless of what filehandle was to begin with -- all we need is for it to respond to `:each`

File: lib/utils/tsv_reader.rb [hunk 2]
```diff
@@ -40,12 +42,12 @@ def records
       # For each line, get a copy of header_hash and populate it using header_index
       # so that cols_hash = {ocn: 11, local_id: i11}
       @inf.each_line do |line|
-        yield line_to_hash(line.strip)
+        yield line_to_hash(line.chomp)
```
Comment (aelkiss) [inline lib/utils/tsv_reader.rb:45]: the changes here are to ensure we can handle trailing empty columns (not sure whether we'd expect to see that in practice or not)

File: spec/clusterable/commitment_spec.rb [hunk 3]
```diff
@@ -19,16 +19,16 @@
   end
 
   it "validates local_shelving_type" do
-    expect(comm.valid?).to be true
+    expect(comm).to be_valid
```
Comment (aelkiss) [inline spec/clusterable/commitment_spec.rb:22]: with all these changes, rspec will better describe failures - rspec can dynamically generate matchers for any predicate ending in `?` that returns a boolean

**Comment by aelkiss** [comment]
Decrease in coverage is primarily filename exception branches in shared print loader - I'm not too worried about that

File: lib/utils/tsv_reader.rb (hunk 2 above)
Comment (mwarin) [reply lib/utils/tsv_reader.rb:45]: yeah that's a change for the better

File: spec/clusterable/commitment_spec.rb (hunk 3 above)
Comment (mwarin) [reply spec/clusterable/commitment_spec.rb:22]: :exploding_head:

File: lib/utils/tsv_reader.rb [hunk 4]
```diff
@@ -40,12 +42,12 @@ def records
...
       end
     end
 
     def line_to_hash(line)
-      cols = line.split(@delim)
+      cols = line.split(@delim, -1)
```
Comment (mwarin) [inline lib/utils/tsv_reader.rb:50]: Why the -1 here? I don't understand.

**Review by mwarin — APPROVED** [review]

**Pull request merged**